src/main/java/org/opensearch/security/filter/SecurityRestFilter.java

```diff
@@ -259,6 +259,17 @@ private boolean checkAndAuthenticateRequest(RestRequest request, RestChannel cha
             }
         }
 
+        if (HTTPHelper.containsOBOToken(request)) {
+            String OBO_ENDPOINT_PREFIX = "_plugins/_security/api/user/onbehalfof";
+            if (request.method() == Method.POST && OBO_ENDPOINT_PREFIX.equals(suffix)) {
```

Comment:
Similar to how the suffixes are configured in [SecurityRestFilter](https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/filter/SecurityRestFilter.java#L89-L90) to skip authc for endpoints ending in the suffix, this could also just be the last portion of the endpoint `onbehalfof`

Reply:
Nice catch. I just fixed.